Mathlib/LinearAlgebra/Matrix/Spectrum.lean

```diff
@@ -34,6 +34,23 @@ namespace IsHermitian
 section DecidableEq
 
 variable [DecidableEq n]
+
+section RankMulUnitaryTheorems
+
+variable {m R : Type*} [CommRing R] [StarRing R]
```

Comment:
Can you please move this section up even higher? For instance, ever before the first `variable` declaration? The issue is that, as written, it leads to a fairly arbitrary hypothesis order for the these declarations:
```
{n : Type*} [Fintype n] [DecidableEq n] {m R : Type*} [CommRing R] [StarRing R]
```
It probably makes more sense to just have a section at the very beginning with `variable {m n R : Type*} ...`

Reply:
Hmm. I'm having issues figuring out where exactly to close this section if I do this, since closing the section will put the variables out of scope. Let me think about the details of why we want a section here...

Reply:
The easiest thing seems to be just to enclose the two theorems within the section, and to declare all variables outside the section, but is this silly to do if we don't have variables to scope?

Reply:
I moved the section up, but had to stay in the `DecidableEq` section or it wouldn't compile. I continued to encapsulate all declarations involving `R` in the section, since `R` isn't used outside this section.